src/unix.rs

```diff
@@ -21,13 +21,27 @@ pub struct Acquired {
 }
 
 impl Client {
-    pub fn new(limit: usize) -> io::Result<Client> {
+    pub fn new(mut limit: usize) -> io::Result<Client> {
+        assert!(limit <= 4096, "Can't have limit larger than 4096 since the pipe would block forever when writing to it");
```

Comment:
I think it would be best to not have this be an assert. Other options could include:

* Instead of assert, return an error.
* Perform a non-blocking write and return an error if it returns EWOULDBLOCK.
* Leave the behavior as-is.

I'm wary about having a hardcoded constant here because pipes configuration can vary system-to-system.

Reply:
Done!

Reply:
I used the first solution since I think that is the most reasonable one.

The second one requires the pipe to be set to non-blocking mode and I prefer not to as I afraid it will break other parts of this crate.